x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/agent_policy/components/agent_policy_advanced_fields/space_selector.tsx

```diff
@@ -55,14 +96,21 @@ export const SpaceSelector: React.FC<SpaceSelectorProps> = ({
     return value.map((v) => {
       const existingOption = options.find((opt) => opt.key === v);
 
+      if (existingOption) {
+        return existingOption;
+      }
+      const color = allSpaces.find((space) => space.id === v)?.color;
+      const label = allSpaces.find((space) => space.id === v)?.name ?? v;
```

Comment:
nit: `allSpaces.find((space) => space.id === v)` could be extracted to a constant